### PR TITLE
[google_maps_flutter_web] Adds options for gesture handling and tilt controls.

### PR DESCRIPTION
**Pull request by Rexios80**
See #3258

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart [hunk 1]
```diff
@@ -62,9 +62,20 @@ gmaps.MapOptions _configurationAndStyleToGmapsOptions(
 
   if (configuration.scrollGesturesEnabled == false ||
       configuration.zoomGesturesEnabled == false) {
-    options.gestureHandling = 'none';
+    window.console.warn(
+      '`scrollGesturesEnabled` and `zoomGesturesEnabled` are now ignored on the web.'
+      ' Use `webGestureHandling: WebGestureHandling.none` instead.',
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:67]: Won't this warn people who are setting these for mobile? We shouldn't have warnings that are impossible to remove in normal usage.

Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:67]: Couldn't we just keep the old behavior if `webGestureHandling` is null?

Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:67]: Wouldn't it be possible to want `scrollGesturesEnabled` or `zoomGesturesEnabled` false for mobile but not want `gestureHandling = 'none'`

Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:67]: Right you said only if it's null

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart [hunk 2]
```diff
@@ -62,9 +62,20 @@ gmaps.MapOptions _configurationAndStyleToGmapsOptions(
...
+    );
+  }
+
+  if (configuration.webGestureHandling != null) {
+    options.gestureHandling = configuration.webGestureHandling!.name;
   } else {
```
Comment (ditman) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:73]:
```suggestion
  } else if (configuration.scrollGesturesEnabled == false ||
      configuration.zoomGesturesEnabled == false) {
      // Old behavior
      options.gestureHandling = 'none';
  } else {
    options.gestureHandling = WebGestureHandling.auto.name;
  }
```

/cc @stuartmorgan is this what you mean? (and remove the warn?)

Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:73]: Yes, exactly (unless there's a reason that won't work, it seems like it's the most backwards-compatible behavior).

**Review by ditman — APPROVED** [review]
LGTM!

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**